internal/api/grpc/resources/action/v3alpha/integration_test/query_test.go

```diff
@@ -216,16 +216,25 @@ func TestServer_GetTarget(t *testing.T) {
 				err := tt.args.dep(tt.args.ctx, tt.args.req, tt.want)
 				require.NoError(t, err)
 			}
-			got, getErr := instance.Client.ActionV3Alpha.GetTarget(tt.args.ctx, tt.args.req)
-			if tt.wantErr {
-				assert.Error(t, getErr, "Error: "+getErr.Error())
-			} else {
-				assert.NoError(t, getErr)
-				wantTarget := tt.want.GetTarget()
-				gotTarget := got.GetTarget()
-				integration.AssertResourceDetails(t, wantTarget.GetDetails(), gotTarget.GetDetails())
-				assert.Equal(t, wantTarget.GetConfig(), gotTarget.GetConfig())
+			retryDuration := 5 * time.Second
+			if ctxDeadline, ok := isolatedIAMOwnerCTX.Deadline(); ok {
+				retryDuration = time.Until(ctxDeadline)
 			}
+
```

Comment:
Setting the retry duration to the context deadline is a bad idea and caused me many headaches debugging other packages. What can happen here is:

1. A tests might fail consistently
2. Testify framework keeps executing retries, test keeps failing.
3. The limit of `retryDuration` might be approached, but not yet passed.
4. There might not enough time left in the context to complete the request.
5. The next call fails on context timeout.
6. `assert.NoError` only sees the `context.DeadlineExceeded` because the request failed. There is no response.
7. `require.EventuallyWithT` will now only print the context error. The actual reason why the test failed is lost.

The `retryDuration` should always be less then the context `Deadline`. At least by the amount an API call might take in the worst case.